lib/opts.nit

```diff
@@ -147,12 +147,25 @@ abstract class OptionParameter
 	protected fun convert(str: String): VALUE is abstract
 
 	# Is the parameter mandatory?
-	var parameter_mandatory: Bool = true is writable
+	var parameter_mandatory = true is writable
 
-	redef fun read_param(it)
+	redef fun read_param(opts, it)
 	do
 		super
-		if it.is_ok and (it.item.is_empty or it.item.chars.first != '-') then
+
+		var ok = it.is_ok
+		if ok and not it.item.is_empty and it.item.chars.first == '-' then
+			# The next item may look like a known command
+			for opt in opts.options do
+				if opt.names.has(it.item) then
+					# The next item is a known command
+					ok = false
+					break
+				end
+			end
+		end
```

Comment:
I think that optional parameters could only be given with a concatenated argument: `-svalue` for the short option `-s` and `--long=value` for the long option `--long`. But currently, there is no way to do that (and this could the the object of a future PR).

Anyway, I'm fine with any approach that is an improvement (so +1)  but I think that the `mandatory_parameter` is currently ill-specified. As you said, there is no real client of the current behavior so fell free do do as you wish. API or behavior change that break non-existing code is not an issue imho.